packages/nextjs-contentful-renderer/src/getPageContent/getPageContent.ts

```diff
@@ -19,18 +20,21 @@ export const getPageContent = async (
 
   const existingPageId = getVisitedPageIdFromCookies(context, slug);
   if (existingPageId) {
-    const page = await getPageById(apolloClient, existingPageId, preview);
+    const { modelData, pageId } = existingPageId;
+    const page = await getPageById(apolloClient, pageId, preview, modelData);
     if (page) {
       return page;
     }
   }
 
-  const route = await getRouteBySlug(apolloClient, slug, preview, domain);
-  if (!route) return undefined;
+  const replicaRoute = await getRouteBySlug(apolloClient, slug, preview, domain);
+  if (!replicaRoute) return undefined;
 
-  const pageId = getPageId(route, context.query.utm_campaign);
-  const page = await getPageById(apolloClient, pageId, preview);
-  setCookie(context, slug, pageId);
+  const modelData = Object.keys(replicaRoute).includes('modelData') ? (replicaRoute as Replica).modelData : null;
```

Comment:
there is a more idiomatic way of doing this check, see [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) in the documentation